### PR TITLE
Remove extra parameter to match constructor of Listener class

### PR DESCRIPTION
**Pull request by ortibazar**
I could not find suitable issue to refer. It is required to create issue to this small changes.

### CODE REVIEW

**Comment by patriknw** [comment]
good catch, no need to create issue for such small thing (in docs), I set the milestone on the PR

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
OK TO TEST

**Comment by patriknw** [comment]
```
[info] LoggingDocSpec:
Coroner Thread Count starts at 18 in docs.event.LoggingDocSpec
[info] - use a logging actor (1 millisecond)
[info] - use a MDC logging actor (2 milliseconds)
[info] - use a MDC logging actor by mixin (3 milliseconds)
[info] - allow registration to dead letters *** FAILED *** (5 milliseconds)
[info]   java.lang.IllegalArgumentException: no matching constructor found on class docs.event.LoggingDocSpec$$anonfun$5$$anon$2$Listener for arguments []
```

@ortigali ah, this is not easy to see at first, but `class Listener` is defined as an inner class, and such takes `this` as parameter even though it's not visible in source.

Very confusing for the reader of the docs. The way to do it is to define them as proper top level classes:

```scala
object LoggingDocSpec {
      //#deadletters
      import akka.actor.{ Actor, DeadLetter, Props }

      class DeadLetterListener extends Actor {
        def receive = {
          case d: DeadLetter => println(d)
        }
      }
      //#deadletters
}

  "allow registration to dead letters" in {
    new AnyRef {
      //#deadletters

      val listener = system.actorOf(Props(classOf[DeadLetterListener]))
      system.eventStream.subscribe(listener, classOf[DeadLetter])
      //#deadletters
    }
  }
```

File: akka-docs/rst/scala/code/docs/event/LoggingDocSpec.scala [hunk 1]
```diff
@@ -134,67 +161,44 @@ class LoggingDocSpec extends AkkaSpec {
   }
 
   "allow registration to dead letters" in {
-    new AnyRef {
```
Comment (ortigali) [inline akka-docs/rst/scala/code/docs/event/LoggingDocSpec.scala]: Removed extra anonymous class. I think that it is needed not to publish LoggingDocSpec instance by using **this** keyword in the `Props(classOf[Listener], this)`

**Review by patriknw — APPROVED** [review]
second LGTM

**Pull request merged**